### PR TITLE
Task 2 completed - Hanan Basheer (20b030018)

### PR DESCRIPTION
**Pull request by dancinglightning**
I have made a few more changes to my task 2 as I felt it needed a bit more polishing and also gradient descent wasn't converging since my learning rate wasn't small enough. Pls merge my PR .

### CODE REVIEW

**Comment by Karrthik-Arya** [comment]
The box plots are still wrong but max scaling definitely helps converge the loss as well as prevents the weights from going to infinity.
